src/components/common/delete-widget/delete-widget.directive.coffee

```diff
@@ -0,0 +1,20 @@
+module = angular.module('impac.components.common.delete-widget',[])
+
+module.directive('commonDeleteWidget', ($templateCache, ImpacWidgetsSvc, ImpacUtilities) ->
+  return {
+    restrict: 'A',
+    template: $templateCache.get('common/delete-widget.tmpl.html'),
+    scope: {
+      parentWidget: '='
+      onDismiss: '&'
+    }
+    link: (scope, element) ->
+      scope.loading = false
+
+      scope.deleteWidget = ->
+        scope.loading = true
+        ImpacWidgetsSvc.delete(scope.parentWidget)
```

Comment:
@cesar-tonnoir I think @alexnoox agrees with me judging by the reaction.. I was going to ask for his input. The `delete-widget` should just be a presentational component. If you apply what you have done to all components, you will find that your codebase makes requests to APIs from all sorts of places. It seems logically for it to encapsulate the request as it's purpose it to delete. But it actually makes much more sense for the widget.component to be the "smart component", and for it to make the request to the widget service. 

Also, think about having many components like this, you will then have the WidgetsSvc dependency across many different places, and it becomes hard to follow.  

You can read up on this design pattern by searching "smart components dumb components" or "smart components presentational components" (e.g http://blog.angular-university.io/angular-2-smart-components-vs-presentation-components-whats-the-difference-when-to-use-each-and-why/).